compiler/rustc_mir_build/src/check_tail_calls.rs

```diff
@@ -131,11 +131,24 @@ impl<'tcx> TailCallCkVisitor<'_, 'tcx> {
         }
 
         {
+            // `#[track_caller]` affects the ABI of a function (by adding a location argument),
+            // so a `track_caller` can only tail call other `track_caller` functions.
+            //
+            // The issue is however that we can't know if a function is `track_caller` or not at
+            // this point (THIR can be polymorthic, we may have an unresolved trait function).
```

Comment:
```suggestion
            // this point (THIR can be polymorphic, we may have an unresolved trait function).
```

Or you can say "generic" here if you want it to be understandable to more general audience :3